### PR TITLE
Feature/dockerize ii

### PR DESCRIPTION
**Pull request by nikolai-b**
Docker with bundle in a volume.

### CODE REVIEW

**Comment by KGibb8** [comment]
As far as I can see this works in a development environment. @sadeden is there more work that needs to be done for production / deployment before merging into dev branch?

File: docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 
 services:
-  db:
+  postgres:
```
Comment (nikolai-b) [inline docker-compose.yml]: I changed this on purpose 
check 
https://github.com/blockades/mmt/pull/35/commits/e5c50c2b04533da2c196a59e970ac73cbc24fad0

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -7,4 +7,5 @@ if [ -f /app/tmp/pids/server.pid ]; then
   rm /app/tmp/pids/server.pid
 fi
 
-docker/wait-for-postgresql.sh bundle exec rails server
+docker/wait-for-postgresql.sh bundle exec rake db:migrate && bundle exec rails server
```
Comment (nikolai-b) [inline docker/entrypoint.sh]: I removed this on purpose, we don't want to migrate every time on load.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+version: '3'
+
+services:
+  postgres:
```
Comment (nikolai-b) [inline docker-compose.yml]: I changed this to db on purpose see
https://github.com/blockades/mmt/pull/35/commits/e5c50c2b04533da2c196a59e970ac73cbc24fad0

File: docker/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+# docker-compose entrypoint
+
+set -e
+
+if [ -f /mmt/tmp/pids/server.pid ]; then
+  rm /mmt/tmp/pids/server.pid
+fi
+
+docker/wait-for-postgresql.sh bundle exec rake db:migrate && bundle exec rails server
```
Comment (nikolai-b) [inline docker/entrypoint.sh]: I removed the migrate on purpose.  We should migrate explicitly

File: docker-compose.yml (hunk 3 above)
Comment (ameba23) [reply docker-compose.yml]: it seems keiran changed it back.  i saw you said that the word 'postgres' is reserved.  but i was having issues with it as it was, see:
https://github.com/blockades/mmt/issues/37

File: docker/entrypoint.sh (hunk 4 above)
Comment (KGibb8) [reply docker/entrypoint.sh]: :+1:

File: docker-compose.yml (hunk 3 above)
Comment (KGibb8) [reply docker-compose.yml]: Apologies, it wasn't working on my machine, did when I changed it to this. Since others had reported same issues.

**Comment by KGibb8** [comment]
Are we okay to merge this? I undid the bad stuff... :kissing_smiling_eyes:

**Comment by nikolai-b** [comment]
Sorry if my comments were abrupt, I just wanted to check if you added the stuff intentionally (its not necessarily _bad_, just that I didn't think it was best practise, I could be wrong though...)

I'll try to double check the docker setup with a clean start soon and report back ASAP

**Comment by ameba23** [comment]
when u do this, please document the exact commands you used to get it working if they do not match those that keiran put in the readme.   (im still having problems getting it running)

**Comment by nikolai-b** [comment]
@ameba23 I've done that.  You might need to wipe old containers and volumes to get a fresh build
```
docker-compose rm -svf
```

Let me know if it works for you!

**Comment by ameba23** [comment]
everything runs fine until: 
```
docker-compose run --rm rake db:create db:migrate setup
```
which gives 
```
ERROR: No such service: rake
```

**Comment by KGibb8** [comment]
Stuff it seems was an accidental push. Git seems to have pushed all branches when I pushed a different one... Annoying.

**Comment by KGibb8** [comment]
I'm back to getting `FATAL:  database "db" does not exist`. BAH.

**Comment by KGibb8** [comment]
Anyone against me merging this?

**Comment by nikolai-b** [comment]
I'd merge this but

screenshot from 2018-01-17 21-25-35

**Comment by KGibb8** [comment]
Settings say you're invited as an admin @nikolai-b... Check your invites?

File: README.md [hunk 5]
```diff
@@ -72,10 +72,10 @@ Development Environment
 docker-compose build
 # this bundle step shouldn't be needed but it is...
 docker-compose run --rm mmt bundle
-docker-compose run --rm rake db:create db:migrate setup
+docker-compose run --rm mmt rake db:create db:migrate setup
```
[the remark this answers is not in the document]
Comment (KGibb8) [reply README.md]: :smile:

**Pull request merged**